### PR TITLE
Add otlpjson as a serializer implementation

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: internal/otlpjson/marshaler.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	buf := bytes.Buffer{}
+	if err := en.delegate.Marshal(&buf, model.(*otlpcollectorlogs.ExportLogsServiceRequest)); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (en *marshaler) MarshalMetrics(model interface{}) ([]byte, error) {
+	buf := bytes.Buffer{}
+	if err := en.delegate.Marshal(&buf, model.(*otlpcollectormetrics.ExportMetricsServiceRequest)); err != nil {
```
Comment (jrcamp) [inline internal/otlpjson/marshaler.go:57]: Do we want these to be panics if the wrong type is passed in? See here for how it checks the type:

https://github.com/open-telemetry/opentelemetry-collector/pull/3242/files#diff-99f0341981a504978dfba25f8c54cbaff15a4ef2022f7e4b72ee35fd25835787R46

It should _probably_ be caught with tests if wrong type is passed but I'm not 100% sure how these API's will be used which is why I went the safer route.

Comment (jrcamp) [reply internal/otlpjson/marshaler.go:57]: Though I was kind of thinking for OTLP the model would still be pdata. (ie the translation step would be a nop, passing through pdata) which the marshaler would then unwrap to the protobuf before serializing.

Comment (jrcamp) [reply internal/otlpjson/marshaler.go:57]: nm, this probably makes more sense.

Comment (bogdandrutu) [reply internal/otlpjson/marshaler.go]: Done.

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Out of curiosity: is this expected to be working eventually in the very same way as `internal/otlptext`? ;)

**Comment by bogdandrutu** [comment]
@carlosalberto `internal/otlptext` is expected to work like this.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**